src/main/scala/com/databricks/spark/sql/perf/results.scala

```diff
@@ -149,8 +149,8 @@ class MLParams(
    */
   def toMap: Map[String, String] = {
     val allParams = ReflectionUtils.getConstructorArgs(this)
-    allParams.map { case (key: String, value: Any) =>
-      key -> value.toString
+    allParams.map { case (key: String, value: Option[Any]) =>
+      key -> value.get.toString
```

Comment:
This will throw an exception if the value is `None` or `null`. Try this:

~~~scala
// Only outputs params that have values.
allParams.flatMap {
  case (key: String,   Some(value: Any)) =>
    Some(key -> value.toString)
  case _ =>
    None
}
~~~